File: Android/app/src/main/java/com/denarii/android/util/InputPatternValidator.java
package com.denarii.android.util;

import android.widget.EditText;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class InputPatternValidator {

  public static boolean isValidInput(EditText text, String patternStr) {
    Pattern pattern = Pattern.compile(patternStr);
    return isValidInput(text, pattern);
  }

  public static boolean isValidInput(EditText text, Pattern pattern) {
    Matcher matcher = pattern.matcher(text.getText().toString());
    return matcher.matches();
  }
}

File: Ios/denarii/denarii/validators/InputPatternValidator.swift
//
//  InputPatternValidator.swift
//  denarii
//
//  Created by Andrew Katson on 1/18/24.
//

import Foundation


func is_valid_input(_ text: String, _ pattern: Regex<Substring>) -> Bool {
    return text.contains(pattern)
}

func is_valid_input(_ text: String, _ pattern: Regex<(Substring, Substring?)>) -> Bool {
    return text.contains(pattern)
}
